src/dialogs/preferences/PreferenceDialog.jsx

```diff
@@ -0,0 +1,51 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+
+/* global window */
+
+const PreferenceDialog = (props) => {
+    const [storage, setStorage] = useState('');
+
+    useEffect(() => {
+        init();
+    }, []);
+
+    function init() {
+        // note: this is an ugly hack required because window.Wikicite is only set after the window has loaded, and is thus not available when the PreferenceDialog is first rendered
```

Comment:
Could't you pass `Wikicite` in instead of `props` to have access to the Wikicite object? Or is there another reason why you're waiting for it here?

Reply:
If I am not mistaken, this will not work because the Wikicite object that is being passed through window.arguments is different from the one that is being set to window.Wikicite (which happens in bootstrap.js:117 on window load). The first one is defined in wikicite.js and only has utility functions like getting localized strings, while the second one is defined in bootstrap.js and has the Prefs property to manage preferences. I do not know why these two different Wikicite objects exist, but I did not want to make any changes to them that could potentially break other stuff.

Maybe a cleaner solution for the preferences dialog would be to wait for window.Wikicite to be set within dialogs/preferences/index.js and only then render PreferencesDialog, passing in window.Wikicite. This would at least externalize the weird waiting logic from the PreferencesDialog React component. What do you think?

Reply:
Thanks for working on this, Robert!

I see the `window.Wikicite` object may not be ready in the preferences window when `PreferenceDialog` is first rendered. But wouldn't it be available in the `wikiciteChrome.js`' `openPreferences` function which launches the preferences dialog? Would passing the `window.Wikicite.Prefs` in the fourth argument of `window.openDialog` ([`wikiciteChrome.js:44`](https://github.com/diegodlh/zotero-cita/blob/75250fcdc8a61d4f9c395831dc4cf74b37a68bc7/src/wikiciteChrome.js#L44)), along with `Wikicite`, work?

Sorry if I'm completely missing the point here. To be honest, I haven't been working on this code recently.

Reply:
> Would passing the window.Wikicite.Prefs in the fourth argument of window.openDialog ([wikiciteChrome.js:44](https://github.com/diegodlh/zotero-cita/blob/75250fcdc8a61d4f9c395831dc4cf74b37a68bc7/src/wikiciteChrome.js#L44)), along with Wikicite, work?

Yes that's indeed a much better and simpler solution. I've updated the code to do this instead. Thanks!